src/posix/client.cpp

```diff
@@ -128,14 +129,44 @@ int main(int argc, char *argv[])
         strncpy(sockname.sun_path, OPENTHREAD_POSIX_DAEMON_SOCKET_NAME, sizeof(sockname.sun_path) - 1);
 
         ret = connect(sSessionFd, reinterpret_cast<const struct sockaddr *>(&sockname), sizeof(struct sockaddr_un));
+    }
 
-        if (ret == -1)
-        {
-            fprintf(stderr, "OpenThread daemon is not running.\n");
-            ExitNow(ret = OT_EXIT_FAILURE);
-        }
+exit:
+    if (ret == -1 && sSessionFd != -1)
+    {
+        close(sSessionFd);
+        sSessionFd = -1;
+    }
+    return ret;
+}
+
+static bool ReconnectSession(void)
+{
+    bool    ok       = false;
+    clock_t deadline = clock() + 3 * CLOCKS_PER_SEC;
+
+    while (clock() <= deadline)
```

Comment:
2 points on this line - 1) you should include "&& !ok". and 2) there's an issue if clock() is less than 3 seconds from wrapping when setting deadline.  In that case you won't enter the while loop because deadline will have been set to a value less than clock().

Reply:
> 1) you should include "&& !ok".

Actually it not necessary to check `!ok` in while because `VerifyOrExit(ConnectSession() == -1, ok = true);` sets `ok` to true and `goto exit`. So, we always have `!ok` satisfied in the loop. 

> 2) there's an issue if clock() is less than 3 seconds from wrapping when setting deadline. In that case you won't enter the while loop because deadline will have been set to a value less than clock().

Good catch! I think we should use `std::chrono` utilities instead. Any suggestions?

Reply:
> > 1. you should include "&& !ok".
> 
> Actually it not necessary to check `!ok` in while because `VerifyOrExit(ConnectSession() == -1, ok = true);` sets `ok` to true and `goto exit`. So, we always have `!ok` satisfied in the loop.
> 
Oh right.  Its not necessary, nice!

> > 1. there's an issue if clock() is less than 3 seconds from wrapping when setting deadline. In that case you won't enter the while loop because deadline will have been set to a value less than clock().
> 
> Good catch! I think we should use `std::chrono` utilities instead. Any suggestions?

I was thinking some like:

Reply:
changed to just use retry. please check.

Reply:
> changed to just use retry. please check.

LGTM